src/main/java/com/mraof/minestuck/client/gui/DialogueScreen.java

```diff
@@ -106,16 +187,33 @@ public void render(GuiGraphics guiGraphics, int mouseX, int mouseY, float partia
 		
 		guiGraphics.blit(dialogueData.guiBackground(), xOffset, yOffset, 0, 0, GUI_WIDTH, GUI_HEIGHT);
 		
+		renderAnimation(guiGraphics);
+		
 		int pY = yOffset + 20;
 		for(FormattedCharSequence line : messageLines)
 		{
 			guiGraphics.drawString(font, line, xOffset + 10, pY, 0x000000, false);
-			pY += 9;
+			pY += DialogueButton.TEXT_SPACING;
 		}
 		
 		super.render(guiGraphics, mouseX, mouseY, partialTicks);
 	}
 	
+	/**
+	 * Places a talksprite directly to the right of the gui. The center height of the sprite is level with that of the gui.
+	 */
+	private void renderAnimation(GuiGraphics guiGraphics)
+	{
+		ResourceLocation sprite = animationData.getRenderPath(dialogueData.spriteType());
+		float scale = animationData.scale();
+		int moddedWidth = (int) (animationData.spriteWidth() * scale);
+		int moddedHeight = (int) (animationData.spriteHeight() * scale);
+		
+		//if there is a .png.mcmeta file associated with the sprite, the animation for it is updated here
+		AnimatableTexture.setAndUpdate(sprite, animationTick++);
```

Comment:
Which seems to be related to the dimensions of the png (the irregular size of non-salamander consorts)

Reply:
Alright I think Ive figured it out now. It now appears that it is unrelated to frame dimensions. It will crash specifically if there is a sprite with only one frame. I am a little confused why thats the case since that didnt happen before but that should help narrow down the problem